langchain4j-core/src/main/java/dev/langchain4j/store/memory/chat/FileChatMemoryStore.java

```diff
@@ -0,0 +1,133 @@
+package dev.langchain4j.store.memory.chat;
+
+import dev.langchain4j.data.message.ChatMessage;
+import dev.langchain4j.data.message.ChatMessageDeserializer;
+import dev.langchain4j.data.message.ChatMessageSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+
+import static dev.langchain4j.internal.Exceptions.illegalArgument;
+import static dev.langchain4j.internal.Exceptions.runtime;
+
+/**
+ * Implementation of {@link ChatMemoryStore} that stores state of {@link dev.langchain4j.memory.ChatMemory} (chat messages) a local file.
+ * <p>
+ * This mechanism is more reliable than {@link InMemoryChatMemoryStore}.
+ */
+public class FileChatMemoryStore implements ChatMemoryStore {
+    private static final Logger log = LoggerFactory.getLogger(FileChatMemoryStore.class);
+
+    private final String storageDirectory;
+    private final String fileExtension;
+
+    /**
+     * Constructor for FileChatMemoryStore.
+     *
+     * @param storageDirectory Folder path where msg is stored.
+     * @param fileExtension    File extension to store messages.
+     */
+    public FileChatMemoryStore(String storageDirectory, String fileExtension) {
+        Path directoryPath = Paths.get(storageDirectory);
+        if (Files.exists(directoryPath)) {
+            if (!Files.isDirectory(directoryPath)) {
+                throw illegalArgument("Storage directory path exists but is a file: %s", storageDirectory);
+            }
+        } else {
+            try {
+                Files.createDirectories(directoryPath);
+            } catch (IOException e) {
+                throw runtime("Something is wrong, Failed to create folder %s", storageDirectory);
```

Comment:
When an exception occurs, users will not know what happened the way this is done currently

Reply:
> When an exception occurs, users will not know what happened the way this is done currently

I agree, I will commit new changes later when I have free time.